source/developers/sts-for-operator.rst

```diff
@@ -0,0 +1,170 @@
+2. The MinIO Operator uses the Kubernetes API to check that the JSON Web Token (JWT) associated with the :ref:`service account <minio-operator-sts-service-account>` in the application's request is valid.
+3. The Kubernetes API returns the results of its validity check.
+4. The MinIO Operator checks for a :ref:`Policy Binding <minio-operator-sts-policy-binding>` that matches the application.
+5. The PolicyBinding CRD returns the policy or policies that match the request, if any.
+6. The MinIO Operator sends the combined policy information for the application to the MinIO Tenant.
+7. The tenant creates temporary credentials matching the policy or policies for the request and returns those to the MinIO Operator.
+8. The MinIO Operator forwards the temporary credentials back to the application.
+9. The application uses the credentials to send the object storage calls to the MinIO tenant.
+
+Requirements
+------------
+
+STS for the MinIO Operator requires the following:
+
+- MinIO Operator v5.0.0 or later.
+- The deployment **must** have :ref:`TLS configured <minio-tls>`.
+- :envvar:`OPERATOR_STS_ENABLED` environment variable set to ``on``.
+
+Procedure
+---------
+
+1. Enable STS functionality for the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl -n minio-operator set env deployment/minio-operator OPERATOR_STS_ENABLED=on
+   
+   - Replace ``minio-operator`` with the namespace for your deployment.
+   - Replace ``deployment/minio-operator`` with the value for your deployment's MinIO Operator.
+
+     You can find the deployment value by running ``kubectl get deployments -n <namespace>``, where you replace ``<namespace>`` with the namespace for the MinIO Operator.
+     Your MinIO Operator namespace is typically ``minio-operator``, though this value can change during install.
+
+2. Ensure an appropriate :ref:`policy <minio-policy>` exists on the MinIO Tenant for the application to use
+
+3. Create a yaml document to add necessary resources: 
+
+   - The :ref:`Service Account <minio-operator-sts-service-account>` in the MinIO Tenant for the application to use.
+
+     For more on service accounts in Kubernetes, see the :kube-docs:`Kubernetes documentation <reference/access-authn-authz/service-accounts-admin/>`.
+   - Create a :ref:`Policy Binding <minio-operator-sts-policy-binding>` linking the application to the MinIO Tenant's policy.
+
+5. Apply the yaml file to create the resources on the deployment
+   
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl apply -k path/to/yaml/file.yaml
+
+6. Use an SDK that supports the ``AssumeRoleWithWebIdentity`` like behavior to send a call from your application to the deployment
+
+   The STS API expects a JWT for the service account to exist in the Kubernetes environment.
+   This is typically in a well known location on the deployment, such as ``/var/run/secrets/kubernetes.io/serviceaccount/token``.
```

Comment:
Maybe is useful mention that if you define the environment variable
`AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/kubernetes.io/serviceaccount/token` then you don't need to read the JWT token file manually and pass it to the SDK client as parameter.

For example AWS SDK describes it here:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list-AWS_WEB_IDENTITY_TOKEN_FILE

On the minio Go SDK we hare an example in the tests

https://github.com/minio/minio-go/blob/8c4c0768b1b1d29b52457d8e048144a8e2901009/pkg/credentials/iam_aws_test.go#L353-L355

not sure how much detailed or extended should be in the docs that mention, but for sure we can have code examples of this on the Operator repo, and just mention "oh you can do AWS_WEB_IDENTITY_TOKEN_FILE too" or somenthing like that.

Reply:
Added some info too this effect in this step of the procedure.